llvm/test/CodeGen/AMDGPU/wwm-regalloc-error.ll

```diff
@@ -0,0 +1,17 @@
+; RUN: not llc -mtriple=amdgcn-amd-amdhsa -mcpu=gfx900 -mattr=-flat-for-global -stress-regalloc=3 -filetype=null %s 2>&1 | FileCheck %s
+
+; A negative test to capture the expected error when the VGPRs are insufficient for wwm-regalloc.
+
+; CHECK: error: can't find enough VGPRs for wwm-regalloc
+
+define amdgpu_kernel void @test() #0 {
+  call void asm sideeffect "", "~{v[0:1]}" ()
+  %val0 = call i32 asm sideeffect "; def $0", "=s" ()
+  %val1 = call i32 asm sideeffect "; def $0", "=s" ()
+  store i32 10, ptr addrspace(1) null
+  call void asm sideeffect "; use $0", "s"(i32 %val0)
+  call void asm sideeffect "; use $0", "s"(i32 %val1)
+  ret void
+}
+
+attributes #0 = { nounwind "amdgpu-num-vgpr"="2"}
```

Comment:
Drop the attribute, we should remove it eventually and it should be redundant with -stress-regalloc

Reply:
The `stress-regalloc` is a niche way to restrict only the AllocationOrder. It is visible only to RegisterClassInfo.cpp design and not useful to constrain the register numbers elsewhere. So, `getMaxNumVGPRs` doesn't incorporate it at all. I should otherwise clobber all available VGPRs via inline ASM to make them unavailable.

Reply:
I've been thinking of moving it to the stress-regalloc to MRI so that targets can access them using a helper function throughout their codegen. But that is altogether a different patch.